src/Raven.Server/Documents/Sharding/Handlers/Batches/ShardedBatchCommand.cs

```diff
@@ -107,6 +99,28 @@ public IEnumerator<SingleShardedCommand> GetEnumerator()
         }
     }
 
+    private void ModifyId(ref BatchRequestParser.CommandData cmd, bool isServerSideIdentity)
+    {
+        if (isServerSideIdentity)
+            cmd.Id = ShardHelper.GenerateStickyId(cmd.Id, _databaseContext.IdentityPartsSeparator); // generated id is 'users/$BASE26$/'
+
+        if (string.Empty == cmd.Id)
+            cmd.Id = Guid.NewGuid().ToString();
+    }
+
+    private int GetShardNumberForCommandType(BatchRequestParser.CommandData cmd, bool isServerSideIdentity)
+    {
+        if (isServerSideIdentity)
+        {
+            return _databaseContext.GetShardNumber(_context, cmd.Id.AsSpan(0, cmd.Id.Length - 2));   // here we are cheating by cutting '$/' from the end to detect shard number based on BASE26 part
```

Comment:
I don't understand what you are doing here. Can this be moved to `ShardHelper` as well?